dff/connectors/messenger/telegram/interface.py

```diff
@@ -0,0 +1,146 @@
+
+    :param messenger: Messenger instance.
+    :param update: Update to process.
+    """
+    if update.update_id > messenger.last_update_id:
+        messenger.last_update_id = update.update_id
+
+    update_fields = vars(update).copy()
+    update_fields.pop("update_id")
+    inner_update = next(filter(lambda val: val is not None, list(update_fields.values())))
+
+    ctx_id = getattr(vars(inner_update).get("from_user"), "id", None)
+    return inner_update, ctx_id
+
+
+class PollingTelegramInterface(PollingMessengerInterface):
+    """
+    Asynchronous Telegram interface that retrieves updates by polling.
+    Multi-threaded polling is currently not supported, but will be implemented in the future.
+
+    :param messenger: :py:class:`~dff.connectors.messenger.telegram.messenger.TelegramMessenger` instance.
+    :param interval: Polling interval. See `link <https://github.com/eternnoir/pyTelegramBotAPI#telebot>`_ .
+    :param allowed_updates: Processed updates. See `link <https://github.com/eternnoir/pyTelegramBotAPI#telebot>`_ .
+    :param timeout: General timeout. See `link <https://github.com/eternnoir/pyTelegramBotAPI#telebot>`_ .
+    :param long_polling_timeout: Polling timeout. See `link <https://github.com/eternnoir/pyTelegramBotAPI#telebot>`_ .
+    """
+
+    def __init__(
+        self,
+        messenger: TelegramMessenger,
+        interval: int = 3,
+        allowed_updates: Optional[List[str]] = None,
+        timeout: int = 20,
+        long_polling_timeout: int = 20,
+    ):
+        self.messenger: TelegramMessenger = messenger
+        self.interval: int = interval
+        self.allowed_updates: Optional[List[str]] = allowed_updates
+        self.timeout: int = timeout
+        self.long_polling_timeout: int = long_polling_timeout
+
+    def _request(self) -> List[Tuple[Any, Hashable]]:
+        updates = self.messenger.get_updates(
+            offset=(self.messenger.last_update_id + 1),
+            allowed_updates=self.allowed_updates,
+            timeout=self.timeout,
+            long_polling_timeout=self.long_polling_timeout,
+        )
+        update_list = [extract_telegram_request_and_id(self.messenger, update) for update in updates]
+        return update_list
+
+    def _respond(self, response: List[Context]):
+        for resp in response:
+            self.messenger.send_response(resp.id, resp.last_response)
+
+    def _on_exception(self, e: Exception):
+        logger.error(e)
+        self.messenger._TeleBot__stop_polling.set()
+
+    async def connect(self, callback: PipelineRunnerFunction, loop: Optional[Callable] = None, *args, **kwargs):
+        self.messenger._TeleBot__stop_polling.clear()
+        self.messenger.get_updates(offset=-1)
+
+        await super().connect(
+            callback, loop=loop or (lambda: not self.messenger._TeleBot__stop_polling.wait(self.interval))
+        )
+
+
+class WebhookTelegramInterface(CallbackMessengerInterface):
```

Comment:
Maybe rename to `CallbackTelegramInterface`?

Reply:
I think, that we can keep this name